pkg/server/share_manager.go

```diff
@@ -105,6 +123,18 @@ func (m *ShareManager) Run() error {
 			}
 
 			m.logger.Info("Starting nfs server, volume is ready for export")
+
+			if err := m.getLease(); err != nil {
+				m.logger.WithError(err).Warn("Failed to get lease - no lease renewal will be done")
+			} else {
+				m.logger.Info("Found lease")
+				if err = m.takeLease(); err != nil {
+					m.logger.WithError(err).Warn("Failed to take lease - no lease renewal will be done.")
+				} else {
+					go m.runLeaseRenew()
+				}
+			}
+
```

Comment:
Should we add a new EVN to the share-manager pod like `enableFastFailureDetection` so that when it is enabled, share-manager pod is required to be able to find the lease object and take ownership of the lease object? In this case, if share-manager pod is not able to find the lease object/take ownership, we need to crash the pod to signal that the `enableFastFailureDetection` requirement is not able to fulfilled 

When `enableFastFailureDetection` is disabled, share-manager pod ignore the lease mechanism. Just save the ETCD traffic similar to previous Longhorn version.

Reply:
I suppose we can do this.  I hadn't pictured that a failure to find the lease would be a serious enough deal to crash the share-manager app.  Normally that would mean that a lease-capable share-manager image is running opposite an older longhorn-manager image that doesn't create the lease, which is easily fixed.  It could also mean that lease creation failed, for whatever reason, which would be tougher.

Note that the longhorn-manager code makes the lease regardless of "feature gate" setting `enable-share-manager-fast-failover`.  That only controls whether it checks for staleness.  So the combination of setting with environment variable is
- setting TRUE, evn TRUE:   lease is updated, staleness is checked.
- setting FALSE, evn TRUE:  lease is updated, but never checked.
- setting TRUE, evn FALSE, lease is not updated, so when it is checked, there will be no holder identity, so not stale.
- setting FALSE, evn FALSE, lease is neither updated nor checked.  This would be the default shipping config?

We would just have to be clear in the documentation that both items have to be enabled for fast failover to work.  Is that what you are picturing?

Reply:
Done.  See below.